### ISSUE TITLE
Fix test suite

### ISSUE THREAD

**Issue by 2e0byo** [tamland/python-tidal#153]
As something of a priority we need to fix the test suite, ideally before we merge anything else.  I'll try to get on this this week, unless anyone else wants to.

Once it passes we should add vcr (`pytest-recording`) with sanitising and run it in ci.

FWIW I think 3.9 is probably broken, but we need tests running in ci to be confident.

**Comment by tehkillerbee** [comment]
I agree. Feel free to take a look at it, as you are definitely the most qualified!

**Comment by 2e0byo** [comment]
Tests now pass.   We should run in vcr with pytest-recording, but for now lets not merge anything unless one of the maintainers has run the test suite locally.

I've not fixed tox as I don't like keeping the deps in two places, so `poetry run pytest`.  Credentials will be cached locally.

**Comment by 2e0byo** [comment]
PS didn't add pytest-recording as it's currently broken.   There's an open (abandoned) PR to work against latest VCR which would fix it; might pick that up if nobody else does.

**Comment by 2e0byo** [comment]
Most of the work to get tests running in ci is floating around in [a branch](https://github.com/2e0byo/python-tidal/tree/feat/recorded-tests) on my fork.  If someone has the time feel free to get it working: we need to mock the right properties on Session() (or modify it) so we can start a session up without the internet, but all the rest works as recorded.  I'm not likely to have much time for this in the next few weeks, but I'll get on it if I can.
